src/Psl/Result/unwrapResultOr.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Result;
+
+/**
+ * Unwrap the given Result if it is succeeded or return $default value.
+ *
+ * @param ResultInterface<T> $r
+ * @param F $f
+ *
+ * @return T|F
+ *
+ * @template T
+ * @template F
+ */
+function unwrapResultOr(ResultInterface $r, mixed $f)
```

Comment:
It would be better to introduce `ResultInterface::unwrapOr(mixed $default): mixed` instead of adding a function. This would be in line with how the `Option` class does it.

Reply:
Agree here. this should be a method in `ResultInterface`.

side note: we use `snake_case` for functions :)